### PR TITLE
Fix password reset page: validate model correctness

### PR DESCRIPTION
**Pull request by skofman1**
Also fixed text on page (copy paste issue from "Forgot Password" view.

@maartenba @xavierdecoster @shishirx34 @scottbommarito @ryuyu

### CODE REVIEW

File: src/NuGetGallery/Views/Users/ResetPassword.cshtml [hunk 1]
```diff
@@ -1,35 +1,33 @@
...
-        send instructions to reset your password.
-    </p>
-} else {
-    <h1 class="page-heading">Set Password</h1>
-
-    <p>
-        You can set a NuGet.org account password using the form below.
+}
+
+<h1 class="page-heading">@(ViewBag.Title)</h1>
```
Comment (ryuyu) [inline src/NuGetGallery/Views/Users/ResetPassword.cshtml]: Old title for Reset Password was "Forgot Password". Just an fyi that with this change, this title changes.

File: src/NuGetGallery/Controllers/UsersController.cs [hunk 2]
```diff
@@ -239,6 +239,11 @@ public virtual ActionResult ResetPassword(bool forgot)
             // By having this value present in the dictionary BUT null, we don't put "returnUrl" on the Login link at all
             ViewData[Constants.ReturnUrlViewDataKey] = null;
 
+            if(!ModelState.IsValid)
```
Comment (maartenba) [inline src/NuGetGallery/Controllers/UsersController.cs]: Very nit: needs a space beween if and (

Comment (maartenba) [inline src/NuGetGallery/Controllers/UsersController.cs]: Very nit: needs a space beween if and (

**Comment by ryuyu** [comment]
So if my understanding is correct, we now have the same form to fill out for reset and set password?

File: tests/NuGetGallery.Facts/Controllers/UsersControllerFacts.cs [hunk 3]
```diff
@@ -289,12 +289,30 @@ public class TheResetPasswordAction : TestContainer
                     ConfirmPassword = "pwd",
                     NewPassword = "newpwd"
                 };
-
+
```
Comment (scottbommarito) [inline tests/NuGetGallery.Facts/Controllers/UsersControllerFacts.cs]: very very nit: get rid of this whitespace

File: src/NuGetGallery/Views/Users/ResetPassword.cshtml [hunk 4]
```diff
@@ -1,35 +1,33 @@
...
+    <fieldset class="form">
+        <legend>@(ViewBag.Title)</legend>
+        @Html.AntiForgeryToken()
+        @Html.ValidationSummary(true)
+
+        @Html.EditorForModel()
+
+        <img src="@Url.Content("~/Content/images/required.png")" alt="Blue border on left means required." />
+        <input type="submit" value="Set" title="Set Password" />
+</fieldset>
```
Comment (scottbommarito) [inline src/NuGetGallery/Views/Users/ResetPassword.cshtml]: very nit: tab for this line?

File: src/NuGetGallery/Views/Users/ResetPassword.cshtml (hunk 1 above)
Comment (scottbommarito) [inline src/NuGetGallery/Views/Users/ResetPassword.cshtml]: Yeah I think you probably want to change [this](https://github.com/NuGet/NuGetGallery/pull/3242/files#diff-be4423b925a01f503f41d7f6428c0cb3R3) to...
```
ViewBag.Title = ViewBag.ForgotPassword ? "Forgot Password" : "Set Password";
```

Comment (skofman1) [inline src/NuGetGallery/Views/Users/ResetPassword.cshtml]: I changed it on purpose. Reset password is the correct headline

File: tests/NuGetGallery.Facts/Controllers/UsersControllerFacts.cs (hunk 3 above)
Comment (skofman1) [inline tests/NuGetGallery.Facts/Controllers/UsersControllerFacts.cs]: ha? look at the entire method

**Pull request merged**